plugins/plugins.go

```diff
@@ -497,6 +528,12 @@ func (m *Manager) Init(ctx context.Context) error {
 	})
 
 	if err != nil {
+		if m.stop != nil {
+			done := make(chan struct{})
```

Comment:
I am curious to know - why does `m.stop` have type `chan chan struct{}`, rather than just `chan struct{}`. I have never seen that pattern before. Isn't instantiating and passing a whole channel more expensive than an empty struct?

Reply:
I guess it's a trampoline type thing...

```go
		case done := <-m.stop:
			cancel()
			ticker.Stop()
			done <- struct{}{}
			return
		}
```

I guess that even though writes to unbuffered channels are blocking, apparently we want to wait until `ticker.Stop()` has also completed before we proceed... which raises the question: why do we want erroring out of `New()` to block until `ticker.Stop()` has completed?

Reply:
You're right we do this to make sure we've cleaned up everything. 

> why do we want erroring out of New() to block until ticker.Stop() has completed?

Not sure where we are doing this?

Reply:
> Not sure where we are doing this?

This pattern is used:

```go
	if m.stop != nil {
		done := make(chan struct{})
		m.stop <- done
		<-done
	}
```

It blocks until `ticker.Stop()` has completed unless I am mistaken.

Reply:
We do this when we initialize and stop the manger, not when we create a new manager instance.